compiler/pipes/final-check.cpp

```diff
@@ -318,20 +318,31 @@ static void check_kphp_tracing_func_enter_branch_call(FunctionPtr current_functi
              "kphp_tracing_func_enter_branch() is available only inside functions with @kphp-tracing aggregate");
 }
 
+void raise_error_if_throwable(const std::string &where, const VertexAdaptor<op_callback_of_builtin> &callback) {
+  std::vector<std::string> throws;
+  for (const auto &e : callback->func_id->exceptions_thrown) {
+    throws.emplace_back(e->name);
+  }
+  kphp_error(false, fmt_format("{} should not throw exceptions\n"
+                               "But it throws {}\n"
+                               "Throw chain: {}",
+                               where.c_str(), vk::join(throws, ", "), callback->func_id->get_throws_call_chain()));
+}
+
 void check_register_shutdown_functions(VertexAdaptor<op_func_call> call) {
   auto callback = call->args()[0].as<op_callback_of_builtin>();
   if (!callback->func_id->can_throw()) {
     return;
   }
```

Comment:
Maybe move it inside `raise_error_if_throwable` to fit its name? Now this function always raises error assuming to take throwable callback

Reply:
Fixed